compiler/rustc_interface/src/passes.rs

```diff
@@ -847,6 +848,45 @@ fn analysis(tcx: TyCtxt<'_>, (): ()) -> Result<()> {
         sess.time("check_lint_expectations", || tcx.check_expectations(None));
     });
 
+    if sess.opts.unstable_opts.print_vtable_sizes {
+        let traits = tcx.traits(LOCAL_CRATE);
+
+        for &tr in traits {
+            let name = ty::print::with_no_trimmed_paths!(tcx.def_path_str(tr));
+
+            let entries = tcx.vtable_entries_no_resolve(tr);
```

Comment:
This ICEs in cases where we need normalization to resolve super traits:
```rust
pub trait A<T: B>: AsRef<[T::V]> {}

pub trait B {
    type V;
}
```

Reply:
Yeah, so from reading the code we only monomorphize vtables on demand:
```rust
// rustc_monomorphize::collector
//! The following things also result in LLVM artifacts, but are not collected
//! here, since we instantiate them locally on demand when needed in a given
//! codegen unit:
//!
//! - Constants
//! - VTables
//! - Object Shims
```

Which means that we won't really collect any information that way, because most crates don't contain `dyn` coercions of their traits...

Reply:
I guess we could just do this manually then instead of using `tcx.vtable_entries_no_resolve` and `tcx.vtable_entries`. Just stick the trait into the elaborator, then walk thru the super traits manually and count its methods using `own_existential_vtable_entries` (+ 3 per trait or whatever, and also counting the pointers for supertraits or whatever -- would need some checks)...